lib/trino-metastore/src/main/java/io/trino/metastore/HiveMetastore.java

```diff
@@ -67,6 +67,8 @@ default boolean useSparkTableStatistics()
 
     List<TableInfo> getTables(String databaseName);
 
+    List<String> getTablesWithParameter(String databaseName, String parameterKey, Set<String> parameterValues);
```

Comment:
Above there is `getTable` and `getTables` that return actual objects, this returns list of strings, should it be named `getTableNamesWithParameters` ?

Reply:
`getTablesWithParameter` was here before and I only re-added it so I would prefer to keep the old name.

Reply:
I don't think it matters what the old name was, ideally the name should be something similar to the glue or thrift metastore API that we eventually call.
`getTableNamesWithParameters` is a better name here